echopop/spatial/variogram.py

```diff
@@ -384,5 +901,421 @@ def variogram(
+            for param, value in zip(fit_parameters, _argument.items()):
+                if isinstance(value, tuple):
+                    _parameters[param][key] = value[1]
+                else:
+                    _parameters[param][key] = value
+
+    # Populate lower bounds
+    populate_parameters(lower_bounds, "min", "lower_bounds")
+
+    # Populate upper bounds
+    populate_parameters(upper_bounds, "max", "upper_bounds")
+
+    # Populate starting values
+    if initial_values is None:
+        # ---- Initialize empty list
+        init_values = []
+        # ---- Create empty fixed list
+        fixed_parameters = []
+        # ---- If no initial values are provided, use default values
+        for param in fit_parameters:
+            init_values.append((param, default_variogram_settings[param]))
+    else:
+        # ---- Create copy of `initial_values`
+        init_values = initial_values.copy()
+        # ---- Get the required parameters
+        variogram_args, variogram_function = get_variogram_arguments(model)
+        # ---- Get names
+        arg_names = list(variogram_args.keys())
+        # ---- Drop `distance_lags`
+        arg_names.remove("distance_lags")
+        # ---- Assign a list of parameters that will be kept fixed
+        fixed_parameters = []
+        # ---- Get the initial values present
+        init_parameters = {param for param, value in initial_values}
+        # ---- Find missing values that are contained within the `fit_parameters` list
+        missing_parameters = set(arg_names) - set(init_parameters)
+        # ---- Append default values to `inital_values`
+        if missing_parameters:
+            for param in missing_parameters:
+                fixed_parameters.append(param)
+                init_values.append((param, default_variogram_settings[param]))
+    # ---- And populate
+    populate_parameters(init_values, "value", "initial_values")
+
+    # Populate fixed constants, if any
+    if fixed_parameters:
+        for param in fixed_parameters:
+            _parameters[param]["vary"] = False
+
+    # Initial a Parameters class from the `lmfit` package
+    # ---- Initialize `paramaeters` object
+    parameters = Parameters()
+    # ---- Populate `parameters`
+    for param in _parameters.keys():
+        parameters.add(param, **_parameters[param])
+
+    # Save parameter names that are present
+    # ---- Parameter names
+    optimization_parameters = [
+        "max_fun_evaluations",
+        "cost_fun_tolerance",
+        "solution_tolerance",
+        "gradient_tolerance",
+        "finite_step_size",
+        "trust_region_solver",
+        "x_scale",
+        "jacobian_approx",
+    ]
+    # ---- Parameter values
+    optimization_values = [
+        max_fun_evaluations,
+        cost_fun_tolerance,
+        solution_tolerance,
+        gradient_tolerance,
+        finite_step_size,
+        trust_region_solver,
+        x_scale,
+        jacobian_approx,
```

Comment:
I think you can omit this by passing in `optimization_parameters` as a dict instead of using ** to expand it at input of this function (`create_optimization_options`).